src/Commands/PartCommand.cpp

```diff
@@ -0,0 +1,66 @@
+#include "PartCommand.hpp"
+#include "Replies.hpp"
+#include <string>
+
+PartCommand::PartCommand(std::string source, std::vector<std::string> params,
+                         Client* client) {
+    checkParams(client, params);
+    parseParams(client, params);
+    this->_source = source;
+    this->_params = params;
+    this->_client = client;
+}
+
+PartCommand::~PartCommand() {}
+
+void PartCommand::checkParams(Client* client, std::vector<std::string> params) {
+    if (params.size() < 1) {
+        client->sendMessage(Replies::ERR_NEEDMOREPARAMS(client, "PART"));
+        throw;
+    }
+}
+
+void PartCommand::parseParams(Client* client, std::vector<std::string> params) {
+    std::istringstream iss(params[0]);
+    std::string chanName;
+    size_t i = 0;
+    this->_reason = "";
+
+    while (std::getline(iss, chanName, ',')) {
+        try {
+            _channels[i] = Server::getInstance().findChannel(chanName);
+        } catch (const Server::ChannelNotFoundException()) {
+            client->sendMessage(Replies::ERR_NOSUCHCHANNEL(client, chanName));
```

Comment:
you need to increment it before continuing, this is not a for loop, it'll not auto-increment
(that's why a for loop is better for most of the cases)

Reply:
my bad I'm dumb